src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Addresses.cs

```diff
@@ -31,8 +25,52 @@ internal static List<GatewayIPAddressInformation> ParseGatewayAddressesFromRoute
                     parser.MoveNextOrFail();
                     string gatewayIPHex = parser.MoveAndExtractNext();
                     long addressValue = Convert.ToInt64(gatewayIPHex, 16);
-                    IPAddress address = new IPAddress(addressValue);
-                    collection.Add(new SimpleGatewayIPAddressInformation(address));
+                    if (addressValue != 0)
+                    {
+                        // Skip device routes without valid NextHop IP address.
+                        IPAddress address = new IPAddress(addressValue);
+                        collection.Add(new SimpleGatewayIPAddressInformation(address));
+                    }
+                }
+            }
+
+            return collection;
+        }
+
+        internal static List<GatewayIPAddressInformation> ParseIPv6GatewayAddressesFromRouteFile(List<GatewayIPAddressInformation> collection, string filePath, string interfaceName, long scopeId)
+        {
+            // Columns are as follows (first-line header):
+            // 00000000000000000000000000000000 00 00000000000000000000000000000000 00 00000000000000000000000000000000 ffffffff 00000001 00000001 00200200 lo
+            // +------------------------------+ ++ +------------------------------+ ++ +------------------------------+ +------+ +------+ +------+ +------+ ++
+            // |                                |  |                                |  |                                |        |        |        |        |
+            // 1                                2  3                                4  5                                6        7        8        9        10
+            // 1. IPv6 destination network displayed in 32 hexadecimal chars without colons as separator
+            // 2. IPv6 destination prefix length in hexadecimal
+            // 3. IPv6 source network displayed in 32 hexadecimal chars without colons as separator
+            // 4. IPv6 source prefix length in hexadecimal
+            // 5. IPv6 next hop displayed in 32 hexadecimal chars without colons as separator
+            // 6. Metric in hexadecimal
+            // 7. Reference counter
+            // 8. Use counter
+            // 9. Flags
+            // 10. Device name
```

Comment:
nit: In my ipv6_route file the device name is pre-padded with spaces to right-align the names. Maybe you can show this in the example.

Reply:
nit2: maybe start counting at zero, so the indexing matches with the token array below.

Reply:
Should the `interfaceName` argument be matched with the `Device name` column somewhere?

Reply:
updated.